### PR TITLE
Init Dockerfile and docker-compose setup

### PR DESCRIPTION
**Pull request by unused**
For some dependencies I struggled to setup a local (Fedora Linux) development environment so I created a docker-compose setup. This should help providing a quick, simple and stable way to get the Exercism website running locally for any contributor familiar with Docker.

- [x] Dockerfile to setup a rails environment
- [x] docker-compose configuration for local development environment
- [x] Rake task for exercism setup steps
- [x] Ensure database uses `utf8mb4` character set

Note: The create `server_identity` file and yarn install steps are redundant in the Dockerfile and afterwards `exercism:setup` rake task, necessary because the data is overwritten by the volume mount, but this makes the Dockerfile independent.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,41 +20,59 @@ mysql -e "GRANT ALL PRIVILEGES ON exercism_reboot_test.* TO 'exercism_reboot'@'l
...
-bundle exec rails s
+bundle exec rails s # shorthand for bundle exec rails server
 ```
 
-Something like this will get a working webserver on http://http://lvh.me:3000
+Something like this will get a working webserver on http://lvh.me:3000
 Note: Teams will be avaliable on http://teams.lvh.me:3000
 
-#### On future runs
+#### On Future Runs
```
Comment (kytrinyx) [inline README.md]: We use sentence case for all of our titles.

**Comment by kytrinyx** [comment]
Hi @unused I totally hear where you're coming from.

I'm very hesitant to pull this in, since nobody on the core team is familiar with Docker, and if we accept this patch, it means that we've agreed to support the docker setup, which we are not in a good place to do.

How would you feel about putting the docker setup in a separate project under your own GitHub account, and then we could link to it from the README or Contributing file as a community supplied/supported thing which people are welcome to try if they're interested?

I think the rake task you've suggested makes sense with or without a docker setup, and I'm open to having the database.yml support env vars that fallback to the defaults we use.

**Comment by unused** [comment]
Ahoi @kytrinyx thank you for the quick reply. For sure I'm happy to share a quick setup using Docker for the Exercism project. I'll prepare the repository and follow up with clean PR for the db config and setup task, as well as a very brief usage description.

**Comment by iHiD** [comment]
Thanks @unused. If you pop a PR to the README when you're done, I'll happily merge that :)

**Pull request closed without merging**